File: go/config/Ota.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Ota 对应 config/Ota.py。
type Ota struct {
	*session.ApiGroup
}

// NewOta 创建 Ota API 分组。
func NewOta(s *session.Session) *Ota {
	return &Ota{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 ota/config.xml。
func (o *Ota) Config() (map[string]interface{}, error) {
	return o.S.Get("ota/config.xml", nil, "config")
}

File: src/config/ota.rs
//! Ota config group (`config/Ota.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Ota config group.
pub struct OtaConfig<'a> {
    conn: &'a Connection,
}

impl<'a> OtaConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        OtaConfig { conn }
    }

    /// `ota/config.xml`. OTA configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/ota/config.xml")
    }
}
